src/components/inputModal.tsx

```diff
@@ -1,27 +1,162 @@
-import { Modal, Button } from "react-bootstrap";
-import React from "react";
-import { useState } from "react";
+import { Modal, Button, Form } from "react-bootstrap";
+import React, { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+
+const schema = yup.object().shape({
+  name: yup.string().required("Name is required"),
+  subject: yup.string().required("Subject is required"),
+  marks: yup.number().positive().integer().min(0).required("Marks are required"),
+  grades: yup.string().required("Grades are required"),
+});
 
 interface ModalProps {
   visible: boolean;
   setVisible: Function;
+  isEdit: boolean;
 }
 
 // function InputModal() {
 const InputModal: React.FC<ModalProps> = (props) => {
```

Comment:
Remove unnecessary comments and use meaningful component names. As this is not a generic modal this name is misleading.